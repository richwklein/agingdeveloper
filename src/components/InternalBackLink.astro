---
/**
 * @name InternalBackButton
 * Component used for rendering a link as a button back to another page.
 *
 * @param {Props} props - The properties of the link.
 * @return {Function} an Astro component factory function.
 *
 * @example
 * <InternalBackLink name={"authors"} path={"/author"}} />
 */
import { Icon } from "astro-icon/components";
import InternalLink from "./InternalLink.astro";

interface Props {
  name: string;
  path: string;
  class?: string;
  useViewText?: boolean;
}

const { name, path, class: className, useViewText = true } = Astro.props;
const text = useViewText ? `view all ${name}` : name;
---

<InternalLink
  to={path}
  class:list={[
    "m-0 inline-flex min-w-16 items-center justify-center rounded-md bg-secondary-main py-2 pl-3 pr-4 capitalize text-secondary-contrast hover:bg-secondary-dark focus:bg-secondary-dark",
    className,
  ]}
>
  <Icon name="mdi:chevron-left" class:list={["mr-1"]} />
  {text}
</InternalLink>
