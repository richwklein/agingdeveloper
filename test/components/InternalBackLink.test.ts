import InternalBackLink from '@components/InternalBackLink.astro';
import { experimental_AstroContainer as AstroContainer } from 'astro/container';
import { describe, expect, test } from 'vitest';

describe("InternalBackButton", async () => {

  const render = async (useViewText = true) => {
    const container = await AstroContainer.create();
    return container.renderToString(InternalBackLink, {
      props: {
        name: "authors",
        path: "/author",
        useViewText: useViewText,
      }
    });
  };

  test("that it renders correctly.", async () => {
    const result = await render();
    expect(result).toMatchSnapshot();
  });

  test("that it contains the text.", async () => {
    const result = await render();
    expect(result).toContain("view all authors");
  });

  test("that it can exclude view all.", async () => {
    const result = await render(false);
    expect(result).not.toContain("view all");
  });

  test("that the icons and href are.", async () => {
    const result = await render();
    expect(result).toContain("data-icon=\"mdi:chevron-left\"");
    expect(result).toContain("href=\"/author\"");
  });
});