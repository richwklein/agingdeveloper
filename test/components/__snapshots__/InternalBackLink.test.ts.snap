// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InternalBackButton > that it renders correctly. 1`] = `"<!DOCTYPE html><!--prettier-ignore --><a data-link="internal" target="_self" href="/author" class="m-0 inline-flex min-w-16 items-center justify-center rounded-md bg-secondary-main py-2 pl-3 pr-4 capitalize text-secondary-contrast hover:bg-secondary-dark focus:bg-secondary-dark" data-astro-source-file="/Users/richklein/Development/Projects/agingdeveloper/src/components/InternalLink.astro" data-astro-source-loc="22:90"> <svg width="1em" height="1em" viewBox="0 0 24 24" class="mr-1" data-icon="mdi:chevron-left" data-astro-source-file="/Users/richklein/Development/Projects/agingdeveloper/node_modules/astro-icon/components/Icon.astro" data-astro-source-loc="117:44">  <symbol id="ai:mdi:chevron-left" data-astro-source-file="/Users/richklein/Development/Projects/agingdeveloper/node_modules/astro-icon/components/Icon.astro" data-astro-source-loc="124:28"><path fill="currentColor" d="M15.41 16.58L10.83 12l4.58-4.59L14 6l-6 6l6 6z"/></symbol><use xlink:href="#ai:mdi:chevron-left" data-astro-source-file="/Users/richklein/Development/Projects/agingdeveloper/node_modules/astro-icon/components/Icon.astro" data-astro-source-loc="125:10"></use>  </svg> view all authors</a>"`;
